src/checker.rs

```diff
@@ -47,17 +47,34 @@ pub fn check(rules: &Rules, raw: &str) -> bool {
         return false;
     }
 
-    let mut words = trimmed.split_whitespace();
+    let words = trimmed.split_whitespace();
     let word_count = words.clone().count();
     if word_count < rules.min_word_count
         || word_count > rules.max_word_count
-        || words.any(|word| rules.disallowed_words.contains(
+        || words.clone().any(|word| rules.disallowed_words.contains(
              &word.trim_matches(|c: char| !c.is_alphabetic()).to_lowercase()
            ))
     {
         return false;
     }
 
+    if !rules.stem_separator_regex.is_empty() {
+        let regex: Regex = Regex::new(&rules.stem_separator_regex).unwrap();
+        let mut stems_words: Vec<&str> = vec![];
+        
+        for word in words {
+            let maybe_stem_word: &str = &(regex.split(word).nth(0).unwrap_or(word));
```

Comment:
I think you missed two other things that were in my previous comment, though I should have mentioned them explicitly as well.

```suggestion
            let maybe_stem_word = regex.split(word).next().unwrap_or(word);
```

* The reference is not needed here
* As `.nth(0)` and `.next()` are the same, let's be more idiomatic and use `.next()`

Reply:
Missed the next() and reference removal before... It was also becoming && on the next line...
This way is the last iteration:
````
        for word in words {
            let maybe_stem_word = regex.split(word).next().unwrap_or(word);
            if maybe_stem_word != word {
                stems_words.push(maybe_stem_word);
            }
        }
```